src/main/java/org/dice_research/opal/civet/metrics/ProviderIdentityMetric.java

```diff
@@ -132,134 +104,79 @@ public Integer compute(Model model, String datasetUri) throws Exception {
-				PublisherScore.put("LandingPageScore", 5);
-			else
-				PublisherScore.put("LandingPageScore", 3);
-		
-			// If LandingPage is not there then check for AccessURL in the distribution
-			if (LandingPageStore.size() == 0) {
+			}
+		}
 
-				NodeIterator DistributionIterator = model.listObjectsOfProperty(dataset,DCAT.distribution);
-				while (DistributionIterator.hasNext()) {
+		// If LandingPage is not there then check for AccessURL in the distribution
+		if (publisher_score == 0) {
 
-					TotalNumberOfDistributions++;
+			int number_of_access_url = 0;
+			int number_of_distributions = 0;
 
-					Resource Distribution = (Resource) DistributionIterator.next();
+			NodeIterator distributions = model.listObjectsOfProperty(dataset, DCAT.distribution);
 
-					if (Distribution.hasProperty(DCAT.accessURL)) {
+			while (distributions.hasNext()) {
 
-						if (isValidURL(Distribution.getProperty(DCAT.accessURL).getObject().toString()))
-							AccessURLStore.add(Distribution.getProperty(DCAT.accessURL).getObject().toString());
+				number_of_distributions++;
+				RDFNode distribution = distributions.next();
+				if (distribution.isResource()) {
+					Resource distribution_resource = (Resource) distribution;
+					if (distribution_resource.hasProperty(DCAT.accessURL)
+							&& isValidURL(distribution_resource.getProperty(DCAT.accessURL).getObject().toString())) {
+						number_of_access_url++;
 					}
 				}
 
-				// Calculate a score based on availability of AccessURL and store a score in
-				// PublisherScore.
-				if (AccessURLStore.size() > 0) {
-
-					int TotalPercentageOfAccessURL = (AccessURLStore.size() * 100) / TotalNumberOfDistributions;
-
-					if (TotalPercentageOfAccessURL == 100)
-						PublisherScore.put("AccessURLScore", 5);
-					else if (TotalPercentageOfAccessURL < 100 && TotalPercentageOfAccessURL >= 75)
-						PublisherScore.put("AccessURLScore", 4);
-					else if (TotalPercentageOfAccessURL < 75 && TotalPercentageOfAccessURL >= 50)
-						PublisherScore.put("AccessURLScore", 3);
-					else if (TotalPercentageOfAccessURL < 50 && TotalPercentageOfAccessURL >= 25)
-						PublisherScore.put("AccessURLScore", 2);
-					else if (TotalPercentageOfAccessURL < 25 && TotalPercentageOfAccessURL > 0)
-						PublisherScore.put("AccessURLScore", 1);
-						
-				}
-				else
-					PublisherScore.put("No provider identity information at all", 0);
 			}
+
+			int TotalPercentageOfAccessURL = (number_of_access_url * 100) / number_of_distributions;
+
+			if (TotalPercentageOfAccessURL == 100)
+				publisher_score = 5;
+			else if (TotalPercentageOfAccessURL < 100 && TotalPercentageOfAccessURL >= 75)
+				publisher_score = 4;
+			else if (TotalPercentageOfAccessURL < 75 && TotalPercentageOfAccessURL >= 50)
+				publisher_score = 3;
+			else if (TotalPercentageOfAccessURL < 50 && TotalPercentageOfAccessURL >= 25)
+				publisher_score = 2;
+			else if (TotalPercentageOfAccessURL < 25 && TotalPercentageOfAccessURL > 0)
+				publisher_score = 1;
```

Comment:
Since you are calculating the score for publisher_score = 1 and publisher_score = 2, please add test cases for them as well.

Reply:
Done. I uploaded testcase for 1 and 2 stars.